pkg/gateway/operations/deleteobjects.go

```diff
@@ -63,6 +63,11 @@ func (controller *DeleteObjects) Handle(w http.ResponseWriter, req *http.Request
 		switch {
 		case errors.Is(err, catalog.ErrNotFound):
 			lg.Debug("tried to delete a non-existent object")
+		case errors.Is(err, catalog.ErrRequiredValue) && resolvedPath.Path == "":
```

Comment:
This is risky: `DeleteEntry` will also return `ErrRequiredValue` if the repository or branch names are empty, and here you are possibly suppressing this error. Consider defining a new error type for an empty path.

Reply:
Alt, you can just check if the `path` we pass we parsed is empty and return success after successful get repo/branch.

Reply:
Done.

Reply:
@nopcoder you mean get repo/branch here or in the cataloger?

If it's here, then it's a wasteful repeatable operation I would like to avoid.
If in the underlying cataloger then I'll need to rely or even modify behaviour for the gateway, which I think is suboptimal.